packages/parser/src/language/pie/pie.langium

```diff
@@ -0,0 +1,19 @@
+grammar Pie
+import "../common/common";
+
+entry Pie:
+  NEWLINE*
+  "pie" showData?="showData"?
+  (
+    NEWLINE* TitleAndAccessibilities sections+=PieSection*
+    | NEWLINE+ sections+=PieSection+
+    | NEWLINE*
+  )
+;
+
+PieSection:
+  label=PIE_SECTION_LABEL ":" value=PIE_SECTION_VALUE NEWLINE+
+;
+
+terminal PIE_SECTION_LABEL: /"[^"]+"/;
+terminal PIE_SECTION_VALUE returns number: /(0|[1-9][0-9]*)(\.[0-9]+)?/;
```

Comment:
I was trying to write grammar for https://github.com/mermaid-js/mermaid/issues/948, and it was working without adding the multiple `NEWLINE*` that's present here. 

Was there any specific reason behind having to explicitly write the newlines?


https://langium.org/playground?grammar=OYJwhgthYgBACmAxgawKYBcBQW0DsMQBPBZdDALi1lgCIAHMzW2AOQFEB1AGQEkOAVNVgAKAEYAbAPaoAzgGoAvACFpqAJQCA3DlUyUVGjVkYYGRfwAqogOQBaG7HwATC60vqA-LBsVHEsDE0CUUAZUsAJX4AcR0sAAsAS2dnfFgMNBAIRLwwCVhOUIpYAHoAHQUSnQysnLy2Lj4OYvKQTzK8Kqwa7Nz8q1gQTABXEDxZWDxhiCCQFoBtAAY7AE4AXXkunrr88KjWaJbaeYA9WjWBWgAfG1ObC5suoA&content=A4Qwxg1gpgLgUABgFwAIBEALKAbbB7FAJkLTgEYBaAZlTQHc8AnbAE1LmOoBYb0AVKAGcYggJYA7AOZFAfBuABvZJA

Reply:
As I said, it won't be possible unless we allowed undesired grammar.

> I don't think so, unless we broke or allowed something that shouldn't be allowed.
> I don't think it's even possible in jison.
>
> https://github.com/mermaid-js/mermaid/pull/4799#discussion_r1313964781

In your code, you used `\s+` as a hidden rule, which means ignore all whitespaces characters, so the following code would be allowed:

```
pie 



showData "section1":
1 "section2":
2
```

Reply:
Another possible approach is to use `EOL`, but `EOF` isn't currently supported: https://github.com/eclipse-langium/langium/pull/1162

```langium
fragment EOL:
  NEWLINE+ | EOF
;
```

> https://github.com/eclipse-langium/langium/pull/1162#issuecomment-1687714247